OneCommunityGlobal/HGNRest
# Jinchao fix visibility issues in Team Member Tasks

# Description
This PR solves the 7.c
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/94319381/b6af54e7-3632-4621-8dd3-c273cfa4e4d6)
The overall visibility logic in Team Member Tasks is same to leaderboard:
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/94319381/327a1d51-aced-4f5a-a1ea-ae4a1b6efe04)
This PR will cover the visibility part and no-team user part. Another PR will be raised to solve the weeklyCommitTime logic.
## Main change
This PR aims to improve the taskController by following the best practice of separating the concerns of controllers and helpers. The controller functions should only handle req and res, while any supporting or business logic should be moved to helper functions. To achieve this, we have refactored the getTasksForTeamsByUser function in the controller by moving the related Mongoose code to a new function called getTasksForTeamsByUser in the taskHelper. Additionally, a new function called getTasksForSingleUser was created to handle cases where the user is not part of any team. This change has simplified the controller function and improved its readability and maintainability.
## How to test
Please test the visibility of Team Member Tasks for different classes of users by creating a team with all classes of users and logging in as each class. Make sure the visibility rules outlined in the specifications are followed. Additionally, test the three buttons to ensure that this PR does not break Kaixiang's latest update: [#800](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/800)
